src/server/routes/consents.ts

```diff
@@ -128,188 +130,296 @@ const getUserNewsletterSubscriptions = async ({
-				payload: consents,
 				request_id,
-			});
-		},
+				accessToken,
+			}),
+			previousPage: 'communication',
+		};
 	},
-	{
-		page: 'review',
-		path: '/consents/review',
-		pageTitle: CONSENTS_PAGES.REVIEW,
-		read: async ({ ip, sc_gu_u, geo, request_id, accessToken }) => {
-			const ALL_CONSENT = [
-				...CONSENTS_DATA_PAGE,
-				...(ConsentsOnNewslettersPageMap.get(geo) as string[]),
-				...CONSENTS_COMMUNICATION_PAGE,
-			];
+	update: async ({ ip, sc_gu_u, accessToken, body, geo, request_id }) => {
+		const userNewsletterSubscriptions = await getUserNewsletterSubscriptions({
+			newslettersOnPage: ALL_NEWSLETTER_IDS,
+			ip,
+			sc_gu_u,
+			accessToken,
+			request_id,
+		});
+
+		// get a list of newsletters to update that have changed from the users current subscription
+		// if they have changed then set them to subscribe/unsubscribe
+		const newsletterSubscriptionsToUpdate =
+			newslettersSubscriptionsFromFormBody(body).filter((newSubscription) => {
+				// find current user subscription status for a newsletter
+				const currentSubscription = userNewsletterSubscriptions.find(
+					({ id: userNewsletterId }) => userNewsletterId === newSubscription.id,
+				);
+
+				// check if a subscription exists
+				if (currentSubscription) {
+					if (
+						// previously subscribed AND now wants to unsubscribe
+						(currentSubscription.subscribed && !newSubscription.subscribed) ||
+						// OR previously not subscribed AND wants to subscribe
+						(!currentSubscription.subscribed && newSubscription.subscribed)
+					) {
+						// then include in newsletterSubscriptionsToUpdate
+						return true;
+					}
+				}
 
-			return {
-				page: 'review',
-				consents: await getUserConsentsForPage({
-					pageConsents: ALL_CONSENT,
-					ip,
-					sc_gu_u,
-					request_id,
-					accessToken,
-				}),
-				newsletters: await getUserNewsletterSubscriptions({
-					newslettersOnPage: NewsletterMap.get(geo) as string[],
-					ip,
-					sc_gu_u,
-					request_id,
-					accessToken,
-				}),
-			};
-		},
+				// otherwise don't include in the update
+				return false;
+			});
+
+		await patchNewsletters({
+			ip,
+			sc_gu_u,
+			accessToken,
+			payload: newsletterSubscriptionsToUpdate,
+			request_id,
+		});
+
+		const consents = (ConsentsOnNewslettersPageMap.get(geo) as string[]).map(
+			(id) => ({
+				id,
+				consented: getConsentValueFromRequestBody(id, body),
+			}),
+		);
+
+		await patchConsents({
+			ip,
+			sc_gu_u,
+			accessToken,
+			payload: consents,
+			request_id,
+		});
 	},
-];
+};
+export class ConsentPages {
+	pages: ConsentPage[];
+	inSimplifyRegistrationFlowTest = () => {
+		const isInABTestVariant = this.ab.isUserInVariant(
+			abSimplifyRegistrationFlowTest.id,
+			abSimplifyRegistrationFlowTest.variants[0].id,
+		);
+		return isInABTestVariant;
+	};
+
+	constructor(readonly ab: ABTestAPI) {
+		if (this.inSimplifyRegistrationFlowTest()) {
+			this.pages = [OUR_CONTENT, YOUR_DATA(OUR_CONTENT.page), REVIEW];
```

Comment:
I think for legibility, it might be easier to extract these two arrays into named variables or a Map so you can just match on the key, and so it's very clear what's going on at a glance. Especially if we ever roll out permanent flow personalistation.

Reply:
yeah good idea, updated with a comment too